src/ShopRepository.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Shopware\ServiceBundle;
+
+use Doctrine\Persistence\ManagerRegistry;
+use Generator;
+use Shopware\ServiceBundle\Entity\Shop;
+
+class ShopRepository
+{
+    public function __construct(private readonly ManagerRegistry $registry) {}
+
+    /**
+     * @return Generator<Shop>
+     */
+    public function findAll(): Generator
+    {
+        $repository = $this->registry->getRepository(Shop::class);
+
+        $batchCount = 100;
+
+        // current offset to navigate over the entire set
+        $offset = 0;
+
+        while ($shops = $repository->findBy([], null, $batchCount, $offset)) {
+            /** @var Shop[] $shops */
```

Comment:
is that necessary? the doctrine code looks like it does support generic annotations

Reply:
ill check again but it was complaining

Reply:
yep you're correct.Not needed